gobblin-salesforce/src/main/java/org/apache/gobblin/salesforce/SalesforceSource.java

```diff
@@ -277,11 +261,13 @@ private List<WorkUnit> createWorkUnits(
   }
 
   /**
-   *
+   * Generates {@link WorkUnit}s based on a bunch of config values like max number of partitions, early stop,
+   * dynamic partitioning, dynamic probing, etc.
    */
-  private List<WorkUnit> generateWorkUnitsStrategy(SourceEntity sourceEntity, SourceState state, long previousWatermark) {
-    Boolean disableSoft = state.getPropAsBoolean(SOURCE_QUERYBASED_SALESFORCE_IS_SOFT_DELETES_PULL_DISABLED, false);
-    log.info("disable soft delete pull: " + disableSoft);
+  @VisibleForTesting
+  List<WorkUnit> generateWorkUnitsHelper(SourceEntity sourceEntity, SourceState state, long previousWatermark) {
+    boolean isSoftDeletePullDisabled = state.getPropAsBoolean(SOURCE_QUERYBASED_SALESFORCE_IS_SOFT_DELETES_PULL_DISABLED, false);
+    log.info("disable soft delete pull: " + isSoftDeletePullDisabled);
```

Comment:
I may not understand the intent behind `SfConfig`... but I expected this to be encapsulated within

Reply:
Yes, the current state is a not great - we have some of these configs in SfConfig, some keys defined in SalesforceConfigurationKeys and some as class constants. 
Will take sorting this up as part part of a separate PR.